data_structures/src/chain.rs

```diff
@@ -331,6 +341,50 @@ impl TryInto<Secp256k1_PublicKey> for PublicKey {
     }
 }
 
+impl From<Secp256k1_SecretKey> for SecretKey {
+    fn from(secp256k1_sk: Secp256k1_SecretKey) -> Self {
+        let mut bytes: [u8; 32] = [0; 32];
+        bytes.copy_from_slice(&secp256k1_sk[..]);
+
+        SecretKey { bytes }
+    }
+}
+
+impl TryInto<Secp256k1_SecretKey> for SecretKey {
```

Comment:
**(1)** `TryInto` can be safely replace by just an `Into`, because both formats are equivalent and correct. If `secp256k1::SecretKey::from_slice` returns an error I think is perfectly ok to just panic since it is something that shouldn't have happened in the first place